src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -279,45 +296,69 @@ public Response modifyClusterSeed(
     LOG.info("modify cluster called with: cluster_name = {}, seedHost = {}", clusterName, seedHost.get());
 
     Optional<Cluster> cluster = context.storage.getCluster(clusterName);
+
     if (!cluster.isPresent()) {
       return Response.status(Response.Status.NOT_FOUND)
           .entity("cluster with name " + clusterName + " not found")
           .build();
     }
 
-    Set<String> newSeeds = parseSeedHosts(seedHost.get());
+    Cluster newCluster = updateClusterSeeds(cluster.get(), seedHost);
 
-    if (context.config.getEnableDynamicSeedList()) {
-      try {
-        JmxProxy jmxProxy =
-            context.jmxConnectionFactory.connectAny(
-                Optional.absent(),
-                newSeeds
-                    .stream()
-                    .map(
-                        host ->
-                            Node.builder()
-                                .withClusterName(clusterName)
-                                .withHostname(parseSeedHost(host))
-                                .build())
-                    .collect(Collectors.toList()),
-                context.config.getJmxConnectionTimeoutInSeconds());
+    if (cluster.get().getSeedHosts().equals(newCluster.getSeedHosts())) {
+      // No change in the node list compared to storage
+      return Response.notModified()
+          .entity(
+              "Topology hasn't changed in cluster " + clusterName + ". No update was performed.")
+          .build();
+    }
+
+    return viewCluster(newCluster.getName(), Optional.<Integer>absent(), Optional.<URI>absent());
+  }
+
+  /**
+   * Updates the list of nodes of a cluster based on the current topology.
+   *
+   * @param cluster the Cluster object we intend to update
+   * @param seedHosts a list of hosts to connect to in the cluster
+   * @return the updated cluster object with a refreshed seed list
+   * @throws ReaperException Any runtime exception that could be triggered
+   */
+  private Cluster updateClusterSeeds(Cluster cluster, Optional<String> seedHosts)
+      throws ReaperException {
+    Set<String> newSeeds = parseSeedHosts(seedHosts.get());
```

Comment:
This seems to assume that seedHosts isn't absent (and will fail if it is). Why are we taking an Optional? For stronger type safety, the responsibility of unwrapping should be in the caller of this function, both of which already do an `.isPresent()` check.

Reply:
That's right, I'll fix this.